pkg/controller/jenkins/configuration/user/seedjobs/seedjobs.go

```diff
@@ -320,9 +333,28 @@ func (s SeedJobs) createAgent(jenkinsClient jenkinsclient.Jenkins, k8sClient cli
 
 	err = k8sClient.Create(context.TODO(), deployment)
 	if apierrors.IsAlreadyExists(err) {
+		currentDeployment := &appsv1.Deployment{}
+		err = k8sClient.Get(context.TODO(), types.NamespacedName{Namespace: jenkinsManifest.Namespace, Name: agentDeploymentName(*jenkinsManifest, agentName)}, currentDeployment)
+		if err != nil {
+			return err
+		}
+
 		err := k8sClient.Update(context.TODO(), deployment)
 		if err != nil {
-			return stackerr.WithStack(err)
+			return err
+		}
+
+		customResourceSeedJobExecutors, err := strconv.Atoi(currentDeployment.Annotations["seedJobExecutorsAmount"])
```

Comment:
if someone puts wrong value when it will break and will it be clearly visible to the user what the problem is?